include/lo2s/trace/trace.hpp

```diff
@@ -114,6 +118,8 @@ class Trace
     void update_process_name(pid_t pid, const std::string& name);
     void update_thread_name(pid_t tid, const std::string& name);
 
+    void mark_process_end(pid_t pid, int status_code);
```

Comment:
Right now `mark_process_begin` is private, since it's only called internally by `add_process`, while `mark_process_end` needs to be public. So that's the reason they're not grouped together, but I admit that this is not optimal. Note that the public and private interfaces of `update_{process,thread}_name` are not grouped together, too.

Do you have a suggestion how we could group those methods more logically?